src/utils.ts

```diff
@@ -71,17 +85,18 @@ export function intersectMany<T>(arrays: T[][]): T[] {
     }
   }
 
-  const set = new Map();
+  const set: Map<string, [number, number]> = new Map();
   for (const elem of arrays[0]) {
-    set.set(elem, 1);
+    set.set(elem[0] ?? "", [1, elem[1]]);
```

Comment:
I think this is not needed. Since the elem is in the array, how can it be `undefined`?
Same applies on line 97 and 99: the `?? ""` are not needed.

Reply:
Let me try :)